api/src/services/import-export.ts

```diff
@@ -155,38 +159,77 @@ export class ImportService {
 			};
 
 			return new Promise<void>((resolve, reject) => {
-				stream
-					.pipe(Papa.parse(Papa.NODE_STREAM_INPUT, PapaOptions))
-					.on('data', (obj: Record<string, unknown>) => {
-						// Filter out all undefined fields
-						for (const field in obj) {
-							if (obj[field] === undefined) {
-								delete obj[field];
-							}
+				const streams: Stream[] = [stream];
+
+				const cleanup = (destroy = true) => {
+					if (destroy) {
+						for (const stream of streams) {
+							destroyStream(stream);
 						}
+					}
 
-						saveQueue.push(obj);
-					})
-					.on('error', (err: any) => {
-						destroyStream(stream);
-						reject(new InvalidPayloadError({ reason: err.message }));
-					})
-					.on('end', () => {
-						// In case of empty CSV file
-						if (!saveQueue.started) return resolve();
+					tmpFile.cleanup().catch(() => {
+						logger.warn(`Failed to cleanup temporary import file (${tmpFile.path})`);
+					});
+				};
 
-						saveQueue.drain(() => {
-							for (const nestedActionEvent of nestedActionEvents) {
-								emitter.emitAction(nestedActionEvent.event, nestedActionEvent.meta, nestedActionEvent.context);
-							}
+				saveQueue.error((error) => {
+					reject(error);
+				});
 
-							return resolve();
+				const fileWriteStream = createWriteStream(tmpFile.path)
+					.on('error', (error) => {
+						cleanup();
+						reject(new Error('Error while writing import data to temporary file', { cause: error }));
+					})
+					.on('finish', () => {
+						const fileReadStream = createReadStream(tmpFile.path).on('error', (error) => {
+							cleanup();
+							reject(new Error('Error while reading import data from temporary file', { cause: error }));
 						});
+
+						streams.push(fileReadStream);
+
+						fileReadStream
+							.pipe(Papa.parse(Papa.NODE_STREAM_INPUT, PapaOptions))
+							.on('data', (obj: Record<string, unknown>) => {
+								// Filter out all undefined fields
+								for (const field in obj) {
+									if (obj[field] === undefined) {
+										delete obj[field];
+									}
+								}
+
+								saveQueue.push(obj);
```

Comment:
Just an improvement to prevent empty lines between records to be imported

```suggestion
								if (Object.keys(obj).length) {
									saveQueue.push(obj);
								}
```